src/model/document.ts

```diff
@@ -268,14 +280,21 @@ export class Document implements Feature {
   }
 
   /**
-   * Adds and indexes a feature to this documentled before resolve().
+   * Return all warnings on local features contained within the document as well
+   * as warnings on features detected by following through all imports. If the
+   * deep argument is set to false, only return warnings found on local
+   * features.
    */
-  _addFeature(feature: Feature) {
-    if (this._doneResolving) {
-      throw new Error('_addFeature can not be called after _resolve()');
+  getWarnings(deep?: boolean): Warning[] {
+    const warnings: Warning[] = [];
+    if (deep == null) {
+      deep = true;
     }
-    this._indexFeature(feature);
-    this._localFeatures.add(feature);
+    warnings.push(...this.warnings);
```

Comment:
oops, forgot that TypeScript won't compile `...` out. Tests are failing on Travis for Node v4 because of these.